src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -1019,15 +1029,27 @@ bool clang_cpp_convertert::name_param_and_continue(
   assert(id.empty() && name.empty());
 
   const clang::DeclContext *dcxt = pd.getParentFunctionOrMethod();
-  if(is_cpyctor(dcxt) && is_defaulted_ctor(dcxt))
+  if(const auto *md = llvm::dyn_cast<clang::CXXMethodDecl>(dcxt))
   {
-    // let's deal with the unnamed parameter for an implicit defaulted cpyctor
-    get_cpyctor_name(
-      static_cast<const clang::CXXConstructorDecl *>(dcxt), id, name, param);
-    return true;
+    if(
+      (is_CopyOrMoveOperator(*md) && md->isImplicit()) ||
+      (is_ConstructorOrDestructor(*md) && is_defaulted_ctor(*md)))
+    {
```

Comment:
this if condition looks weird, can you explain it?

Reply:
Constructors and destructors do not have `md->isImplicit()`, so it failed a lot of TCs because `assert(md->isImplicit())` when I applied the patch, so I made some changes to pass TCs.